staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -578,6 +579,47 @@ func (w *watchCache) Resync() error {
 	return nil
 }
 
+func (w *watchCache) currentCapacity() int {
+	w.Lock()
+	defer w.Unlock()
+	return w.capacity
+}
+
+const (
+	// minWatchChanSize is the min size of channels used by the watch.
+	minWatchChanSize = 3
```

Comment:
I would not change min size till we have proper tracking of terminating watchers in load test, WDYT?

Reply:
I'm fine with blocking on perf-test change on that.

I wanted to change that, because technically we're risking increasing some of channels and wanted to compensate for that.

Reply:
Issue created for it: https://github.com/kubernetes/perf-tests/issues/2054